[BUG] File upload fails: no such file or directory for some PDFs

### Description

I am trying to import/process a few PDFs which continiously fail with the error message "file or directory not found". The PDFs are pretty old (<2019). Other PDFs work without any issues. Also the old PDFs work when "printing them as PDF" (so basically create a new PDF from it)

upon request I can share some of the PDFs (they contain PI, so I cannot share them by default). 

If there is anythign I can look up without sharing the PDF, pls let me know.

### Steps to reproduce

1. take a (old) pdf and upload it via drag & drop

it fails with the error message
`201801_PDF_Rechnung_M211180192221249_01-2018.pdf: The following error occurred while storing document 201801_PDF_Rechnung_M211180192221249_01-2018.pdf after parsing: [Errno 2] No such file or directory: '/tmp/paperless/paperless-ngxqfhcl0wz/uo/201801_PDF_Rechnung_M211180192221249_01-2018.pdf'`

### Webserver logs

```bash
[2024-11-02 19:44:08,852] [DEBUG] [paperless.tasks] Skipping plugin CollatePlugin

[2024-11-02 19:44:08,852] [DEBUG] [paperless.tasks] Skipping plugin BarcodePlugin

[2024-11-02 19:44:08,853] [DEBUG] [paperless.tasks] Executing plugin WorkflowTriggerPlugin

[2024-11-02 19:44:08,854] [INFO] [paperless.tasks] WorkflowTriggerPlugin completed with:

[2024-11-02 19:44:08,854] [DEBUG] [paperless.tasks] Executing plugin ConsumeTaskPlugin

[2024-11-02 19:44:08,867] [INFO] [paperless.consumer] Consuming 201801_PDF_Rechnung_M211180192221249_01-2018.pdf

[2024-11-02 19:44:08,870] [DEBUG] [paperless.consumer] Detected mime type: application/octet-stream

[2024-11-02 19:44:08,870] [DEBUG] [paperless.consumer] Detected possible PDF with wrong mime type, trying to clean with qpdf

[2024-11-02 19:44:08,878] [INFO] [paperless.consumer] qpdf exited 0

[2024-11-02 19:44:08,879] [DEBUG] [paperless.consumer] Detected mime type after qpdf: application/pdf

[2024-11-02 19:44:08,879] [ERROR] [paperless.consumer] Error attempting to clean PDF: [Errno 2] No such file or directory: '/tmp/paperless/paperless-ngxcq3n78i1/uo/201801_PDF_Rechnung_M211180192221249_01-2018.pdf'

[2024-11-02 19:44:08,883] [DEBUG] [paperless.consumer] Parser: RasterisedDocumentParser

[2024-11-02 19:44:08,885] [DEBUG] [paperless.consumer] Parsing 201801_PDF_Rechnung_M211180192221249_01-2018.pdf...

[2024-11-02 19:44:08,894] [INFO] [paperless.parsing.tesseract] pdftotext exited 0

[2024-11-02 19:44:08,973] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': PosixPath('/tmp/paperless/paperless-ngxcq3n78i1/201801_PDF_Rechnung_M211180192221249_01-2018.pdf'), 'output_file': PosixPath('/tmp/paperless/paperless-lpso_23s/archive.pdf'), 'use_threads': True, 'jobs': '2', 'language': 'deu', 'output_type': 'pdfa', 'progress_bar': False, 'color_conversion_strategy': 'RGB', 'skip_text': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': PosixPath('/tmp/paperless/paperless-lpso_23s/sidecar.txt'), 'invalidate_digital_signatures': True}

[2024-11-02 19:44:09,068] [INFO] [ocrmypdf._pipelines.ocr] Start processing 2 pages concurrently

[2024-11-02 19:44:09,069] [INFO] [ocrmypdf._pipeline] skipping all processing on this page

[2024-11-02 19:44:09,069] [INFO] [ocrmypdf._pipeline] skipping all processing on this page

[2024-11-02 19:44:09,075] [INFO] [ocrmypdf._pipelines.ocr] Postprocessing...

[2024-11-02 19:44:09,216] [INFO] [ocrmypdf._pipeline] Image optimization ratio: 1.00 savings: 0.0%

[2024-11-02 19:44:09,216] [INFO] [ocrmypdf._pipeline] Total file size ratio: 1.42 savings: 29.4%

[2024-11-02 19:44:09,218] [INFO] [ocrmypdf._pipelines._common] Output file is a PDF/A-2B (as expected)

[2024-11-02 19:44:09,226] [DEBUG] [paperless.parsing.tesseract] Incomplete sidecar file: discarding.

[2024-11-02 19:44:09,261] [INFO] [paperless.parsing.tesseract] pdftotext exited 0

[2024-11-02 19:44:09,262] [DEBUG] [paperless.consumer] Generating thumbnail for 201801_PDF_Rechnung_M211180192221249_01-2018.pdf...

[2024-11-02 19:44:09,265] [DEBUG] [paperless.parsing] Execute: convert -density 300 -scale 500x5000> -alpha remove -strip -auto-orient -define pdf:use-cropbox=true /tmp/paperless/paperless-lpso_23s/archive.pdf[0] /tmp/paperless/paperless-lpso_23s/convert.webp

[2024-11-02 19:44:09,806] [INFO] [paperless.parsing] convert exited 0

[2024-11-02 19:44:10,195] [DEBUG] [paperless.consumer] Saving record to database

[2024-11-02 19:44:10,195] [DEBUG] [paperless.consumer] Creation date from parse_date: 2018-01-01 00:00:00+01:00

[2024-11-02 19:44:10,446] [DEBUG] [paperless.matching] DocumentType Rechnung matched on document 2018-01-01 201801_PDF_Rechnung_M211180192221249_01-2018 because it contains this word: rechnung

[2024-11-02 19:44:10,446] [INFO] [paperless.handlers] Assigning document type Rechnung to 2018-01-01 201801_PDF_Rechnung_M211180192221249_01-2018

[2024-11-02 19:44:10,453] [DEBUG] [paperless.matching] Tag Chris matched on document 2018-01-01 201801_PDF_Rechnung_M211180192221249_01-2018 because it contains this word: Christian

[2024-11-02 19:44:10,456] [INFO] [paperless.handlers] Tagging "2018-01-01 201801_PDF_Rechnung_M211180192221249_01-2018" with "Chris"

[2024-11-02 19:44:10,512] [ERROR] [paperless.consumer] The following error occurred while storing document 201801_PDF_Rechnung_M211180192221249_01-2018.pdf after parsing: [Errno 2] No such file or directory: '/tmp/paperless/paperless-ngxcq3n78i1/uo/201801_PDF_Rechnung_M211180192221249_01-2018.pdf'

Traceback (most recent call last):

  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 327, in main_wrap

    raise exc_info[1]

  File "/usr/src/paperless/src/documents/consumer.py", line 567, in run

    self._write(

  File "/usr/src/paperless/src/documents/consumer.py", line 792, in _write

    with open(source, "rb") as read_file, open(target, "wb") as write_file:

         ^^^^^^^^^^^^^^^^^^

FileNotFoundError: [Errno 2] No such file or directory: '/tmp/paperless/paperless-ngxcq3n78i1/uo/201801_PDF_Rechnung_M211180192221249_01-2018.pdf'

[2024-11-02 19:44:10,513] [DEBUG] [paperless.parsing.tesseract] Deleting directory /tmp/paperless/paperless-lpso_23s

[2024-11-02 19:44:10,513] [ERROR] [paperless.tasks] ConsumeTaskPlugin failed: 201801_PDF_Rechnung_M211180192221249_01-2018.pdf: The following error occurred while storing document 201801_PDF_Rechnung_M211180192221249_01-2018.pdf after parsing: [Errno 2] No such file or directory: '/tmp/paperless/paperless-ngxcq3n78i1/uo/201801_PDF_Rechnung_M211180192221249_01-2018.pdf'

Traceback (most recent call last):

  File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 327, in main_wrap

    raise exc_info[1]

  File "/usr/src/paperless/src/documents/consumer.py", line 567, in run

    self._write(

  File "/usr/src/paperless/src/documents/consumer.py", line 792, in _write

    with open(source, "rb") as read_file, open(target, "wb") as write_file:

         ^^^^^^^^^^^^^^^^^^

FileNotFoundError: [Errno 2] No such file or directory: '/tmp/paperless/paperless-ngxcq3n78i1/uo/201801_PDF_Rechnung_M211180192221249_01-2018.pdf'


The above exception was the direct cause of the following exception:


Traceback (most recent call last):

  File "/usr/src/paperless/src/documents/tasks.py", line 148, in consume_file

    msg = plugin.run()

          ^^^^^^^^^^^^

  File "/usr/src/paperless/src/documents/consumer.py", line 621, in run

    self._fail(

  File "/usr/src/paperless/src/documents/consumer.py", line 151, in _fail

    raise ConsumerError(f"{self.filename}: {log_message or message}") from exception

documents.consumer.ConsumerError: 201801_PDF_Rechnung_M211180192221249_01-2018.pdf: The following error occurred while storing document 201801_PDF_Rechnung_M211180192221249_01-2018.pdf after parsing: [Errno 2] No such file or directory: '/tmp/paperless/paperless-ngxcq3n78i1/uo/201801_PDF_Rechnung_M211180192221249_01-2018.pdf'
```


### Browser logs

_No response_

### Paperless-ngx version

2.13.2

### Host OS

Debian x86_64 

### Installation method

Docker - official image

### System status

```json
{
    "pngx_version": "2.13.2",
    "server_os": "Linux-6.8.12-3-pve-x86_64-with-glibc2.36",
    "install_type": "docker",
    "storage": {
        "total": 368766496768,
        "available": 346207088640
    },
    "database": {
        "type": "sqlite",
        "url": "/usr/src/paperless/data/db.sqlite3",
        "status": "OK",
        "error": null,
        "migration_status": {
            "latest_migration": "paperless_mail.0027_mailaccount_expiration_mailaccount_account_type_and_more",
            "unapplied_migrations": []
        }
    },
    "tasks": {
        "redis_url": "redis://broker:6379",
        "redis_status": "OK",
        "redis_error": null,
        "celery_status": "OK",
        "index_status": "OK",
        "index_last_modified": "2024-11-02T19:44:10.503629+01:00",
        "index_error": null,
        "classifier_status": "OK",
        "classifier_last_trained": "2024-11-02T18:05:01.610342Z",
        "classifier_error": null
    }
}
```


### Browser

Chrome

### Configuration changes

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/documents/consumer.py
+++ b/src/documents/consumer.py
@@ -411,6 +411,7 @@ class ConsumerPlugin(
                     self.unmodified_original = (
                         Path(tempdir.name) / Path("uo") / Path(self.filename)
                     )
+                    self.unmodified_original.parent.mkdir(exist_ok=True)
                     copy_file_with_basic_stats(
                         self.input_doc.original_file,
                         self.unmodified_original,

